channel/channel.go

```diff
@@ -26,69 +32,129 @@ func WithExpiration[T any](bufSize int) Timed[T] {
+func (c *Timed[T]) BlockingSendWithContext(ctx context.Context, e T) {
 	select {
 	case c.c <- e:
 		return
 	case <-ctx.Done():
-		Logger.Warn().Msgf("%s %X\n%s", BlockedPush, c.c, string(debug.Stack()))
+		Logger.Warn().Msgf("%s %X\n%s", ErrFailedToSend, c.c, string(debug.Stack()))
 		c.c <- e
-		Logger.Info().Msgf("%s %X", UnblockedPush, c.c)
+		Logger.Info().Msgf("unblocked channel %X on send", c.c)
 	}
 }
 
-// PushWithTimeout adds an element in the channel,
+// BlockingSendWithTimeout adds an element in the channel,
 // or logs a warning if it fails after the given timeout
-func (c *Timed[T]) PushWithTimeout(t time.Duration, e T) {
+func (c *Timed[T]) BlockingSendWithTimeout(t time.Duration, e T) {
 	ctx, cancel := context.WithTimeout(context.Background(), t)
 	defer cancel()
 
-	c.PushWithContext(ctx, e)
+	c.BlockingSendWithContext(ctx, e)
 }
 
-// Push adds an element in the channel,
+// BlockingSend adds an element in the channel,
 // or logs a warning if it fails after default timeout
-func (c *Timed[T]) Push(e T) {
+func (c *Timed[T]) BlockingSend(e T) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultChannelTimeout)
 	defer cancel()
 
-	c.PushWithContext(ctx, e)
+	c.BlockingSendWithContext(ctx, e)
 }
 
-// PopWithContext removes an element from the channel
+// BlockingReceiveWithContext removes an element from the channel
 // or logs a warning if it fails after the given context
-func (c *Timed[T]) PopWithContext(ctx context.Context) T {
+func (c *Timed[T]) BlockingReceiveWithContext(ctx context.Context) T {
 	var e T
 
 	select {
 	case e = <-c.c:
 	case <-ctx.Done():
-		Logger.Warn().Msgf("%s %X\n%s", BlockedPop, c.c, string(debug.Stack()))
+		Logger.Warn().Msgf("%s %X\n%s", ErrFailedToReceive, c.c, string(debug.Stack()))
 		c.c <- e
-		Logger.Info().Msgf("%s %X", UnblockedPop, c.c)
+		Logger.Info().Msgf("unblocked channel %X on receiving", c.c)
 	}
 
 	return e
 }
 
-// PopWithTimeout removes an element from the channel
+// BlockingReceiveWithTimeout removes an element from the channel
 // or logs a warning if it fails after the given timeout
-func (c *Timed[T]) PopWithTimeout(t time.Duration) T {
+func (c *Timed[T]) BlockingReceiveWithTimeout(t time.Duration) T {
 	ctx, cancel := context.WithTimeout(context.Background(), t)
 	defer cancel()
 
-	return c.PopWithContext(ctx)
+	return c.BlockingReceiveWithContext(ctx)
 }
 
-// Pop removes an element from the channel
+// BlockingReceive removes an element from the channel
 // or logs a warning if it fails after the default timeout
-func (c *Timed[T]) Pop() T {
+func (c *Timed[T]) BlockingReceive() T {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultChannelTimeout)
+	defer cancel()
+
+	return c.BlockingReceiveWithContext(ctx)
+}
+
+// NotBlockingSendWithContext adds an element in the channel,
+// or returns an error if it fails in the given context
+func (c *Timed[T]) NotBlockingSendWithContext(ctx context.Context, e T) error {
```

Comment:
Here and in the next few functions... minor English detail 😁 

```suggestion
// NonBlockingSendWithContext adds an element in the channel,
// or returns an error if it fails in the given context
func (c *Timed[T]) NonBlockingSendWithContext(ctx context.Context, e T) error {
```